proto/wippersnapper/digitalio/v1/digitalio.proto

```diff
@@ -0,0 +1,34 @@
+// SPDX-FileCopyrightText: 2023 Brent Rubell, Loren Norman, Tyeth Gundry for Adafruit Industries
+// SPDX-License-Identifier: MIT
+syntax = "proto3";
+
+package wippersnapper.digitalio.v1;
+import "nanopb/nanopb.proto";
+
+// DigitalInput is a digital input pin.
+message DigitalIOAdd {
+    int32 gpio_num = 1; // correlates with `name` in the JSON
+    DigitalIODirection gpio_direction = 2;
+    float period = 3; /** Optionally specifies the time between measurements, in seconds. */
+    bool value = 4; /** For sync only - optionally send the pin's value. */
+
+    /**
+    * Direction. Specifies the pin's direction, INPUT or OUTPUT.
+    */
+    enum DigitalIODirection {
+        DIGITAL_IO_DIRECTION_UNSPECIFIED = 0; /** Invalid Direction from Broker. */
+        DIGITAL_IO_DIRECTION_INPUT       = 1; /** Set the pin to behave as an input. */
+        DIGITAL_IO_DIRECTION_INPUT_PULL_UP = 2; /** Set the pin to behave as an input. */
+        DIGITAL_IO_DIRECTION_OUTPUT      = 3; /** Set the pin to behave as an output. */
```

Comment:
Ok, thanks for clarification. I did not see in the API how periodic vs. event sampling is specified. If it happens through the `period` value, the comment could clarify the different modes.

Event counting would be when you want to know total count, even if each event is not immediately reported - for example  a utility meter counting pulses. But that may be better fit in the Sensor category instead of DigitalIO.

Reply:
> Event counting would be when you want to know total count, even if each event is not immediately reported - for example a utility meter counting pulses. But that may be better fit in the Sensor category instead of DigitalIO.


Yeah, I need to figure out where this fits into the API. I'm refactoring to include the sampling mode as an enum submessage.